sway-lsp/src/server_state.rs

```diff
@@ -148,7 +151,24 @@ impl ServerState {
                             experimental,
                         ) {
                             Ok(_) => {
-                                mem::swap(&mut *session.engines.write(), &mut engines_clone);
+                                if let Ok(path) = uri.to_file_path() {
+                                    let path = Arc::new(path);
+                                    let source_id =
+                                        session.engines.read().se().get_source_id(&path);
+                                    if let Some(metrics) = session.metrics.get(&source_id) {
```

Comment:
What does it mean if there are no metrics for the file?

Reply:
true, this can never happen. metrics are always returned along with the programs from the compiler. I'll turn this into an expect.